core/src/main/scala/kafka/server/LogDirFailureChannel.scala

```diff
@@ -49,6 +49,13 @@ class LogDirFailureChannel(logDirNum: Int) extends Logging {
       offlineLogDirQueue.add(logDir)
   }
 
+  /*
+   * Return whether the given log dir is offline.
+   */
+  def logDirIsOffline(logDir: String): Boolean = {
+    offlineLogDirs.containsKey(logDir)
```

Comment:
I am not sure if it is really necessary, but since offline dirs are a rare situation, I'm wondering if makes sense to optimize for the common case to avoid the lookup. For example, maybe we could leave `offlineLogDirs` uninitialized until the first log dir failure.